mycroft/client/speech/mic.py

```diff
@@ -308,6 +324,36 @@ def stop(self):
         """
         self._stop_signaled = True
 
+    def _upload_file(self, filename):
```

Comment:
This is very loosely connected to the ResponsiveRecognizer and could very well be moved outside the class. Basically config and lock as far as I can see.

Reply:
Not really a critical issue but something to think about for future refactoring